### PR TITLE
Added api to report posts and comments (#197)

### LINKED ISSUE #197: API for reporting posts
As a user, I need to be able to report offensive or spammy posts or comments

[PRAW docs](http://praw.readthedocs.io/en/latest/code_overview/models/submission.html?highlight=report()#praw.models.Submission.report)

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #197

#### What's this PR do?
Adds an api for reporting posts and comments to moderators

#### How should this be manually tested?
Go to /api/v0/reports/ and POST a new report (e.g. `{'comment_id': 'ID', 'reason': 'spam'} and verify in the reddit UI that this shows up for your moderator user under the reports.

### CODE REVIEW

**Comment by aliceriot** [comment]
ok this works! I can report a comment, and if I do it shows up in the moderation UI in reddit. Reading code now.

File: channels/serializers.py [hunk 1]
```diff
@@ -536,3 +536,34 @@ def create(self, validated_data):
...
+            raise ValidationError("You must provide only one of either 'post_id' or 'comment_id', not both")
+
+        return attrs
+
+    def create(self, validated_data):
+        """Create a new report"""
+        api = Api(user=self.context['request'].user)
+        post_id = validated_data.get('post_id', None)
+        comment_id = validated_data.get('comment_id', None)
+        reason = validated_data['reason']
```
Comment (aliceriot) [inline channels/serializers.py]: should we validate that `reason` is included as well?

File: channels/views/reports.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    View to report a comment or post
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get_serializer_context(self):
+        """Context for the request and view"""
+        return {
+            'request': self.request,
+            'view': self,
```
Comment (aliceriot) [inline channels/views/reports.py]: it doesn't look like we're using this `view` property in the serializer anyway - do we need to add it for some behind-the-scenes thing?

File: channels/serializers.py (hunk 1 above)
Comment (rhysyngsun) [reply channels/serializers.py]: serializer fields are required by default in DRF so we shouldn't need to

File: channels/views/reports.py (hunk 2 above)
Comment (rhysyngsun) [reply channels/views/reports.py]: It was just from copy/paste of other views, I'll give removing it a try

Comment (rhysyngsun) [reply channels/views/reports.py]: Removed it.

File: channels/serializers.py (hunk 1 above)
Comment (aliceriot) [reply channels/serializers.py]: oh right, I learned that before but forgot! haha

**Comment by rhysyngsun** [comment]
@aliceriot feedback addressed

**Pull request merged**